src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1466,15 +1455,15 @@ public override int GetHashCode()
         public int Hour => (int)((uint)(UTicks / TimeSpan.TicksPerHour) % 24);
 
         internal bool IsAmbiguousDaylightSavingTime() =>
-            InternalKind == KindLocalAmbiguousDst;
+            _dateData >> KindShift == KindLocalAmbiguousDst >> KindShift;
 
         public DateTimeKind Kind
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get => InternalKind switch
+            get => (_dateData >> KindShift) switch
             {
-                KindUnspecified => DateTimeKind.Unspecified,
-                KindUtc => DateTimeKind.Utc,
+                (int)DateTimeKind.Unspecified => DateTimeKind.Unspecified,
+                (int)DateTimeKind.Utc => DateTimeKind.Utc,
```

Comment:
`DateTimeKind` is public API and the kind constants are implementation detail. There's no reason to remove them.

Reply:
These enum values are directly used when setting the flags value, so they will also be appropriate when reading flags.